addons/web_editor/static/src/js/backend.js

```diff
@@ -405,7 +405,19 @@ var FieldTextHtml = widget.extend({
                 var layoutInfo = this.editor.rte.editable().data('layoutInfo');
                 $.summernote.pluginEvents.codeview(undefined, undefined, layoutInfo, false);
             }
-            this.editor.buildingBlock.clean_for_save();
+            if (this.$iframe.is(':visible') || !$.browser.mozilla) {
+                this.editor.buildingBlock.clean_for_save();
+            } else {
+                // force iframe display while transcoding for firefox
+                var ancestors = this.$iframe.parentsUntil(':visible').addBack().each(function () {
+                    $(this).data('rteSavedStyle', $(this).attr('style') || null);
+                    $(this).css({'display': 'initial', 'visibility': 'hidden', 'height': 1});
+                });
+                this.editor.buildingBlock.clean_for_save();
+                ancestors.each(function () {
+                    $(this).attr('style', $(this).data('rteSavedStyle'));
+                });
+            }
```

Comment:
In fact, I would update the whole code to (with conventions):
```javascript
  // Force iframe display while transcoding (... blablabla details)
  var $ancestors = this.$iframe.parentsUntil(':visible').addBack(':not(:visible)');
  _.each($ancestors, function (el) {
      var $el = $(el);
      $el.data('rteSavedStyle', $el.attr('style') || null);
      $el.css({display: 'initial', visibility: 'hidden', height: 1});
  });
  this.editor.buildingBlock.clean_for_save();
  _.each($ancestors, function (el) {
      var $el = $(el);
      $el.attr('style', $el.data('rteSavedStyle'));
      $el.removeData('rteSavedStyle');
  });
```
(no need for the `if` at all, `_.each`, `removeData`, ...)

Note: untested code.
Note 2: to be complete, I would also avoid using `data` to store values... :)